test/dialogs.test.js

```diff
@@ -30,4 +35,42 @@ test("About dialog is populated with correct values", () => {
   expect(SpreadsheetApp.getUi().showModalDialog).toBeCalledTimes(1);
   expect(SpreadsheetApp.getUi().showModalDialog.mock.calls[0][1]).toBe('About');
 
+});
+
+
+test("Test getdialog Function", () => {
+  // store number of keys in Template mock for reference
+  const mockTemplateKeyCoumt = Object.keys(HtmlService.templateMock).length;
+
+  var getDialog = require('src/dialogs.gs').getDialog;
+  var returnedDialogue = getDialog("path/to/html", {
+    username: "username", // commented 
+    password: "password",
+    var1: "value1",
+    var2: "value2",
+    var3: "value3"
+  });
+  // Correct HTML file loaded
+  expect(HtmlService.createTemplateFromFile).toBeCalled();
+  expect(HtmlService.createTemplateFromFile.mock.calls[0][0]).toBe("path/to/html");
+  // password and username not passed to the template
+  // @TODO: Check if the commented out tests should work/..
+  //expect(HtmlService.templateMock["username"]).not.toBeDefined(); // test fails - bug I think
```

Comment:
@ljay79  - I think this may be a bug? I expected username to be removed from the collection but its not. I know we've talked about this before and I may have the wrong expectation of what the code is supposed to be doing.

Reply:
dialogs.gs
`debug.log('Processing: %s.html with %s', file, JSON.stringify(debugValue));`
Is just removing the values from the debug log message but not from values passed to the dialog, as it is required value in Settings dialog for example.

Reply:
Got it - I've been confused by this before